events/validators.go

```diff
@@ -71,8 +71,49 @@ func validateCloudEvent(name string, gotBytes, wantBytes []byte) error {
 		return fmt.Errorf("unmarshalling expected cloud event %q: %v", name, err)
 	}
 
-	if want.String() != got.String() {
-		return fmt.Errorf("unexpected event %q: got %s, want %s", name, got.String(), want.String())
+	gotContext := got.Context.AsV1()
+	wantContext := want.Context.AsV1()
+
+	fields := []struct {
+		name      string
+		gotValue  interface{}
+		wantValue interface{}
+	}{
+		{
+			name:      "ID",
+			gotValue:  gotContext.ID,
+			wantValue: wantContext.ID,
+		},
+		{
+			name:      "source",
+			gotValue:  gotContext.Source,
+			wantValue: wantContext.Source,
+		},
+		{
+			name:      "type",
+			gotValue:  gotContext.Type,
+			wantValue: wantContext.Type,
+		},
+		{
+			name:      "datacontenttype",
+			gotValue:  *gotContext.DataContentType,
+			wantValue: *wantContext.DataContentType,
+		},
+		{
+			name:      "data",
+			gotValue:  string(got.DataEncoded),
+			wantValue: string(want.DataEncoded),
+		},
+	}
+	for _, field := range fields {
+		if field.gotValue != field.wantValue {
+			return fmt.Errorf("unexpected %q field in %q: got %v, want %v", field.name, name, field.gotValue, field.wantValue)
```

Comment:
If there's an error, this will return early before checking all the fields. How about `cmp.Equal()`?
https://godoc.org/github.com/google/go-cmp/cmp#Equal

Reply:
The issue with a single cmp.Equal is that certain fields are going to need special validation, which is what this is setting up. I can use cmp.Equal to un-special case the time field, but to fix e.g. https://github.com/GoogleCloudPlatform/functions-framework-conformance/issues/8, we're going to need to check e.g. the data field in a more flexible way.